specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json

```diff
@@ -1013,9 +1013,455 @@
+            "schema": {
+              "$ref": "#/definitions/CertificateVerificationDescription"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The body contains the certificate.",
+            "schema": {
+              "$ref": "#/definitions/CertificateDescription"
+            }
+          },
+          "default": {
+            "description": "DefaultErrorResponse",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetails"
+            }
+          }
+        },
+        "deprecated": false
+      }
     }
   },
   "definitions": {
+	"CertificateVerificationDescription":{
+      "description": "The JSON-serialized leaf certificate",
+      "type": "object",
+      "properties": {
+        "certificate": {
+          "description": "base-64 representation of X509 certificate .cer file or just .pem file content.",
+          "type": "string"
+        }
+      }
+    },
+    "CertificateBodyDescription":{
+      "description": "The JSON-serialized X509 Certificate.",
+      "type": "object",
+      "properties": {
+        "certificate": {
+          "description": "base-64 representation of the X509 leaf certificate .cer file or just .pem file content.",
+          "type": "string"
+        }
+      }
+    },
+    "CertificateListDescription": {
+      "description": "The JSON-serialized array of Certificate objects.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The array of Certificate objects.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CertificateDescription"
+          }
+        }
+      }
+    },
+    "CertificateDescription": {
+      "description": "The X509 Certificate.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The name of the certificate.",
+          "type": "string"
+        },
+        "etag": {
+          "description": "The entity tag.",
+          "type": "string"
+        },
+        "subject": {
+          "description": "The certificate's subject name.",
+          "type": "string"
+        },
+        "expiry": {
+          "description": "The certificate's expiration date and time.",
+          "type": "string",
+          "format": "date-time-rfc1123"
+        },
+        "thumbprint": {
+          "description": "The certificate's thumbprint.",
+          "type": "string"
+        },
+        "isVerified": {
+          "description": "Determines wether certificate has been verified.",
+          "type": "boolean"
+        },
+        "created": {
+          "description": "The certificate's create date and time.",
+          "type": "string",
+          "format": "date-time-rfc1123"
+        },
+        "updated": {
+          "description": "The certificate's last update date and time.",
+          "type": "string",
+          "format": "date-time-rfc1123"
+        }
+      },
```

Comment:
is there any properties of this model that server can set but client is not allowed to change/set (e.g. `created` seems a property that service set right?).. could you mark those properties as readonly?